lib/AST/Decl.cpp

```diff
@@ -2708,6 +2708,29 @@ DestructorDecl *ClassDecl::addImplicitDestructor() {
   DD->setBody(BraceStmt::create(ctx, getLoc(), { }, getLoc(), true));
   addMember(DD);
   setHasDestructor();
+
+  // Propagate access control and versioned-ness.
+  DD->copyFormalAccessAndVersionedAttrFrom(this);
+
+  // Wire up generic environment of DD.
+  DeclContext *DC = DD->getDeclContext();
+  GenericEnvironment *env = DC->getGenericEnvironmentOfContext();
+  DD->setGenericEnvironment(env);
+
+  // Assign DD the interface type (Self) -> () -> ()
+  ArrayRef<AnyFunctionType::Param> noParams;
```

Comment:
Both `None` and `{}` will have the same behavior (if used directly as arguments).